### PR TITLE
metrics: improve the region heartbeat metrics

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Sometimes we wanna know why there is an error from region heartbeat metrics. But we don't know the exact reason for the region heartbeat error.

### What is changed and how it works?

This PR distinguishes the error kind for the metrics of region heartbeat.

### Check List

Tests

- Unit test

### CODE REVIEW

File: server/grpc_service.go [hunk 1]
```diff
@@ -418,10 +418,13 @@ func (s *Server) RegionHeartbeat(stream pdpb.PD_RegionHeartbeatServer) error {
 		region := core.RegionFromHeartbeat(request)
 		if region.GetLeader() == nil {
 			log.Error("invalid request, the leader is nil", zap.Reflect("request", request), errs.ZapError(errs.ErrLeaderNil))
+			regionHeartbeatCounter.WithLabelValues(storeAddress, storeLabel, "report", "invalid-leader").Inc()
+			msg := fmt.Sprintf("invalid request leader, %v", request)
+			s.hbStreams.SendErr(pdpb.ErrorType_UNKNOWN, msg, request.GetLeader())
```
Comment (lhy1024) [inline server/grpc_service.go:423]: pdpb.ErrorType_UNKNOWN？

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: server/grpc_service.go (hunk 1 above)
Comment (rleungx) [reply server/grpc_service.go:423]: Yes

**Review by lhy1024 — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**